chat-sdk-firebase-adapter/src/main/java/co/chatsdk/firebase/wrappers/ThreadWrapper.java

```diff
@@ -530,27 +531,35 @@ public Completable deleteThread() {
      * Converting the thread details to a map object.
      **/
     private Map<String, Object> serialize() {
+        Map<String , Object> values = new HashMap<>();
 
-        Map<String , Object> value = new HashMap<String, Object>();
-        Map<String , Object> nestedMap = new HashMap<String, Object>();
+        // Thread Details
+        Map<String , Object> detialsMap = new HashMap<>();
 
-        nestedMap.put(Keys.CreationDate, ServerValue.TIMESTAMP);
+        Object creationDate = model.getCreationDate();
+        if (creationDate == null) creationDate = ServerValue.TIMESTAMP;
```

Comment:
This line is valid but you will need to remove:

```
thread.setCreationDate(new Date());
```

From `FirebaseThreadHandler.java`